code/game/machinery/bank_machine.dm

```diff
@@ -19,52 +23,57 @@
 	radio.set_listening(FALSE)
 	radio.recalculateChannels()
 
+	for(var/datum/bank_account/department/each_account in SSeconomy.budget_accounts)
+		if(!each_account.nonstation_account)
+			list_of_budgets += each_account
+
+
 /obj/machinery/computer/bank_machine/Destroy()
+	end_siphon()
 	QDEL_NULL(radio)
 	. = ..()
 
 /obj/machinery/computer/bank_machine/attackby(obj/item/I, mob/user)
-	var/value = 0
+	var/value
 	if(istype(I, /obj/item/stack/spacecash))
 		var/obj/item/stack/spacecash/C = I
 		value = C.value * C.amount
 	else if(istype(I, /obj/item/holochip))
 		var/obj/item/holochip/H = I
 		value = H.credits
-	if(value)
-		var/datum/bank_account/D = SSeconomy.get_budget_account(ACCOUNT_CAR_ID)
-		if(D)
-			D.adjust_money(value)
-			to_chat(user, span_notice("You deposit [I]. The Cargo Budget is now $[D.account_balance]."))
+	if(!value)
+		return
+
+	if(HAS_TRAIT(SSstation, STATION_TRAIT_UNITED_BUDGET))
+		var/datum/bank_account/united_budget = SSeconomy.get_budget_account(ACCOUNT_CAR_ID)
+		united_budget.adjust_money(value)
+		to_chat(user, span_notice("You deposit [value] into a station budget account.</span>"))
 		qdel(I)
 		return
-	return ..()
 
-/obj/machinery/computer/bank_machine/process(delta_time)
-	..()
-	if(siphoning)
-		if (machine_stat & (BROKEN|NOPOWER))
-			say("Insufficient power. Halting siphon.")
-			end_syphon()
-			ui_update()
-			return
-		var/siphon_am = 100 * delta_time
-		var/datum/bank_account/D = SSeconomy.get_budget_account(ACCOUNT_CAR_ID)
-		if(!D.has_money(siphon_am))
-			say("Cargo budget depleted. Halting siphon.")
-			end_syphon()
-			ui_update()
-			return
-
-		playsound(src, 'sound/items/poster_being_created.ogg', 100, TRUE)
-		syphoning_credits += siphon_am
-		D.adjust_money(-siphon_am)
-		if(next_warning < world.time && prob(15))
-			var/area/A = get_area(loc)
-			var/message = "Unauthorized credit withdrawal underway in [initial(A.name)]!!"
-			radio.talk_into(src, message, radio_channel)
-			next_warning = world.time + minimum_time_between_warnings
+	var/list/budget_choice = list()
+	for(var/datum/bank_account/department/budget as anything in list_of_budgets)
+		budget_choice += budget.department_id
+	budget_choice += "All"
+	var/targeted_budget = tgui_input_list(user, "Into which budget would you like to deposit the money?", "Money deposit", budget_choice)
```

Comment:
This sleeps and the preconditions are not re-checked.
Perform the following actions:
- Click on the bank machine with a large amount of money.
- Before confirming the TGUI input list, alt click the money and take all but 1 credit out.
- Confirm the TGUI input.
- Only the 1 credit will be deleted, but the full amount will be inserted.